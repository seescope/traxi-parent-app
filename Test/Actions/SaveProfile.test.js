import saveProfile from '../../App/Actions/SaveProfile';
import { NativeModules, AsyncStorage } from 'react-native';
import { mockSet } from 'firebase-old';

describe('saveProfile', () => {
  it('saves the profile to AsyncStorage', () => {
    const mockSetItem = jest.fn(() => Promise.resolve());
    const mockRegister = jest.fn(() => Promise.resolve('mock-token'));

    AsyncStorage.setItem = mockSetItem;
    NativeModules.NotificationManager = { register: mockRegister };

    const TEST_KID = { name: 'bob' };
    const TEST_USER_ID = 'arbostrisand';
    const TEST_PROFILE = { UUID: TEST_USER_ID };

    return saveProfile(TEST_PROFILE, TEST_KID).then(() => {
      expect(mockSetItem.mock.calls).toMatchSnapshot();
      expect(mockSet.mock.calls).toMatchSnapshot();
    });
  });
});
