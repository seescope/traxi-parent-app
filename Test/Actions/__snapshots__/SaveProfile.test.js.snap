// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saveProfile saves the profile to AsyncStorage 1`] = `
Array [
  Array [
    "profile",
    "{\\"UUID\\":\\"arbostrisand\\"}",
  ],
]
`;

exports[`saveProfile saves the profile to AsyncStorage 2`] = `
Array [
  Array [
    Object {
      "UUID": "arbostrisand",
    },
    [Function],
  ],
]
`;
